### DISCUSSION TITLE
Cache. images on S3

### DISCUSSION THREAD

**Discussion by zoomphoto** [dietercoopman/smart#18] [Q&A]
Hey there, first, thank you for this awesome package!  I have a question, as it pertains to the caching of images.  Are they just a local cache for that browser?

I am envisioning, that once an image is rendered and pulled, I can save that render to S3, then pull it from S3 at anytime in the future.  This way, the first person to view the page gets the loading time, everyone else can pull from cache.

Is that how it does work, or can?  I can't see anything in the docs.

Thanks!

**Comment by dietercoopman** [comment]
Hi @zoomphoto , 

You have interpreted it very well, the images are only processed the first time they are "requested" , after that they are stored to the Laravel cache that uses the cache as provided in the cache config. So served later from that cache.

I will put some more attention into that use case , I understand what you mean. I don't know if there is a s3 cache driver available in the community but I will search for it.  If we have that available then we can store our cached images on s3 too.


Thx for this nice thought 

Have a nice day ☀️

**Reply by zoomphoto** [reply]
I see, so these here:

https://laravel.com/docs/8.x/cache

I wonder if it would possible to store the md5 of the image in redis, and also a rendered link to the actual file.. that would be cool!  

So the flow could be:

- Image is requested
- Check the md5 in redis to see if it already exists
- If yes, serve that file
- If no, render a new file and serve that while adding that key md5 to redis

That would be killer awesome!

**Reply by zoomphoto** [reply]
This actually looks promising:

https://stackoverflow.com/questions/62612251/laravel-aws-s3-storage-image-cache

```
$value = Cache::remember('my-cache-key', 3600 * $hours, function () use ($hours, $image) {
    $url = Storage::disk('s3')->temporaryUrl(
        "images/".$image->filename, now()->addMinutes(60 * $hours + 1)
    ); 
});
```

The key would be the md5 of the file, but I guess it needs to be rendered to know that md5, so it's almost that each rendered file would have to store the config it used to create it,... hmmm, maybe you have that already, I would have to dig in the code.  I am still learning Laravel, however, I've been using php for over a decade...

**Reply by dietercoopman** [reply]
Hi @zoomphoto , thx for the input , will look into it 👀
